extensions/markdown-language-features/src/languageFeatures/copyFiles/shared.ts

```diff
@@ -96,20 +164,24 @@ export async function getMarkdownLink(document: vscode.TextDocument, ranges: rea
 	return { additionalEdits, label };
 }
 
-function checkPaste(document: vscode.TextDocument, ranges: readonly vscode.Range[], regex: RegExp, index: number): boolean {
-	const rangeStartOffset = document.offsetAt(ranges[index].start);
-	const rangeEndOffset = document.offsetAt(ranges[index].end);
-	const matches = [...document.getText().matchAll(regex)];
-	for (const match of matches) {
-		if (match.index !== undefined && rangeStartOffset > match.index && rangeEndOffset < match.index + match[0].length) {
-			return true;
+export function checkSmartPaste(documentText: string, rangeStartOffset: number, rangeEndOffset: number): SmartPaste {
+	const SmartPaste: SmartPaste = { useDefaultPaste: false, updateTitle: false };
+	for (const regex of smartPasteRegexes) {
+		const matches = [...documentText.matchAll(regex.regex)];
+		for (const match of matches) {
+			if (match.index !== undefined) {
+				SmartPaste.useDefaultPaste = rangeStartOffset > match.index && rangeEndOffset < match.index + match[0].length;
+				SmartPaste.updateTitle = regex.is_markdown_link && rangeStartOffset === match.index && rangeEndOffset === match.index + match[0].length;
+				if (SmartPaste.useDefaultPaste || SmartPaste.updateTitle) {
+					return SmartPaste;
+				}
+			}
 		}
 	}
-
-	return false;
+	return SmartPaste;
 }
 
-export async function tryGetUriListSnippet(document: vscode.TextDocument, urlList: String, token: vscode.CancellationToken, title = '', placeHolderValue = 0, smartPaste = false): Promise<{ snippet: vscode.SnippetString; label: string } | undefined> {
+export async function tryGetUriListSnippet(document: vscode.TextDocument, urlList: String, token: vscode.CancellationToken, title = '', placeHolderValue = 0, useDefaultPaste = false): Promise<{ snippet: vscode.SnippetString; label: string } | undefined> {
```

Comment:
Instead of `useDefaultPaste` (which is a little confusing to me) maybe you could flip this around and have a variable that says `pasteAsMarkdownLink`?